File: complex.py

class complex:
    def __init__(self, r=0, j=1):
        self.r = r
        self.j = j

    def __add__(self, other):
        if isinstance(other, int):
            return complex(self.r + other, self.j)
        elif isinstance(other, complex):
            return complex(self.r + other.r, self.j + other.j)
        else:
            raise TypeError

    def __truediv__(self, other):
        if isinstance(other, int):
            return complex(self.r / other, self.j / other)
        elif isinstance(other, complex):
            return complex((self.r * other.r + self.j * other.j) / (other.r*other.r + other.j*other.j), (self.j * other.r - self.r * other.j) / (other.r*other.r + other.j*other.j))
        else:
            raise TypeError

    def __float__(self):
        return float(self.r) / self.comp

    def __int__(self):
        return self.r / self.j

    def __mul__(self, other):
        if isinstance(other, int):
            return complex(other * self.r, other * self.j)
        elif isinstance(other, complex):
            return complex(((self.r * other.r) - (self.j * other.j)), ((self.j * other.r) + (self.r * other.j)))
        else:
            raise TypeError

    def __radd__(self, other):
        return self + other

    def __rtruediv__(self, other):
        return complex(other / self.r, 0 / self.j)

    def __rmul__(self, other):
        return self * other

    def __rsub__(self, other):
        return complex(other - self.r, 0 - self.j)

    def __str__(self):
        return '(' + str(self.r) + ") + (" + str(self.j) + ')i'

    def __sub__(self, other):
        if isinstance(other, int):
            return complex(self.r - other, self.j)
        elif isinstance(other, complex):
            return complex(self.r - other.r, self.j - other.j)
        else:
            raise TypeError

if __name__ == '__main__':

    a = complex(1, 2)
    b = complex(2, 3)

    i = 5

    
    print('%s + %s = %s' % (a, b, a + b))
    print('%s - %s = %s' % (a, b, a - b))
    print('%s * %s = %s' % (a, b, a * b))
    print('%s / %s = %s' % (a, b, a / b))

    print('%s + %i = %s' % (a, i, a + i))
    print('%s - %i = %s' % (a, i, a - i))
    print('%s * %i = %s' % (a, i, a * i))
    print('%s / %i = %s' % (a, i, a / i))

    print('%i + %s = %s' % (i, a, i + a))
    print('%i - %s = %s' % (i, a, i - a))
    print('%i * %s = %s' % (i, a, i * a))
    print('%i / %s = %s' % (i, a, i / a))

File: complex.java
public class Complex 
{
    public Complex() 
	{
        this(0);
    }

    public Complex(double r) 
	{
        this(r, 1);
    }

    public Complex(double r, double j) 
	{
        this.r = r;
        this.j = j;
    }

    public Complex add(Complex o) 
	{
		return new Complex(o.r + r, o.j + j);
    }

    public Complex add(double n) 
	{
        return new Complex(n + r, j);
    }
    
    public Complex add(double n, Complex o)
    {
    	return new Complex(n + o.r, o.j);
    }

    public Complex div(Complex o) 
	{
        return new Complex((o.r * r + o.j * j) / (o.r*o.r + o.j*o.j), (o.r * j - o.j * r) / (o.r*o.r + o.j*o.j));
    }

    public Complex div(double n) 
	{
        return new Complex(r / n, j / n);
    }
    
    public Complex div(double n, Complex o)
    {
    	return new Complex(n / o.r, 0 / o.r);
    }

    public Complex mul(Complex o) 
	{
        return new Complex(((o.r * r) - (o.j * j)), ((o.j * r) + (o.r * j)));
    }

    public Complex mul(double n) 
	{
        return new Complex(n * r, n * j);
    }
    
   	public Complex mul(double n, Complex o)
   	{
   		return new Complex(n * o.r, n * o.j);
   	}

    public Complex sub(Complex o) 
	{
        return new Complex(r - o.r, j - o.j);
    }

    public Complex sub(double n) 
	{
        return new Complex(r - n, j);
    }
    
    public Complex sub(double n, Complex o)
    {
    	return new Complex(n - o.r, 0 - o.j);
    }

    public String toString() 
	{
        return "(" + r + ") + (" + j + ")i";
    }
    
    private double j;
    private double r;
}
